docs/docs/routing.md

```diff
@@ -0,0 +1,261 @@
+---
+id: routing
+title: Routing
+permalink: docs/routing.html
+---
+
+Routing is the process of choosing what to render based on the browser's current URL.
+
+While URLs are an important part of the web, not all apps need to perform routing. For example, single-screen utilities and games will often be unaware of their URL. 
+
+React itself does not have a built-in routing solution. But if you need one, component [state and lifecycle](/react/docs/state-and-lifecycle.html) are all you need to hook React up to the relevant browser APIs. This guide will take you through the process of building a simple Router using these tools.
+
+There are also a number of community projects, such as [react-router](https://github.com/ReactTraining/react-router), that can handle this task for you.
```

Comment:
@dimitriid we haven't had a breaking change in the API for almost 2 years, history's props are no long spread into the route component props in v4, and named routes *prevent the ability to do dynamic routing and code splitting* (can also be implemented in like 10 lines of code on top)